src/test/java/co/rsk/bitcoinj/core/TransactionWitnessTest.java

```diff
@@ -186,11 +202,231 @@ public void equals_withTwoTransactionWitnessesWithOneDifferentPush_shouldBeFalse
+            BtcTransaction.SigHash.ALL, false);
+
+        // sign with fedKey2
+        signInput(btcTx, fedKey2, FIRST_INPUT_INDEX, btcTxSigHash);
+        TransactionWitness witnessWithSignature = btcTx.getWitness(FIRST_INPUT_INDEX);
+
+        // act & assert
+        int sigIndexForFedKey1 = witnessWithSignature.getSigInsertionIndex(btcTxSigHash, fedKey1);
+        int sigIndexForFedKey3 = witnessWithSignature.getSigInsertionIndex(btcTxSigHash, fedKey3);
+
+        Assert.assertEquals(0, sigIndexForFedKey1);
+        // fedKey3 should be inserted after fedKey2
+        Assert.assertEquals(1, sigIndexForFedKey3);
+
+        // now fedKey1 signs the input and pushes fedKey2's signature one position
+        signInput(btcTx, fedKey1, FIRST_INPUT_INDEX, btcTxSigHash);
+        TransactionWitness witnessWithFedKey1Signature = btcTx.getWitness(FIRST_INPUT_INDEX);
+
+        sigIndexForFedKey3 = witnessWithFedKey1Signature.getSigInsertionIndex(btcTxSigHash, fedKey3);
+        Assert.assertEquals(2, sigIndexForFedKey3);
+    }
+
+    @Test
+    public void getSigInsertionIndex_withFedKey3SignatureInTheWitness_shouldReturnIndexCorrectly() {
+        // arrange
+        BtcTransaction btcTx = getBtcTransactionWithBaseWitnessInInput();
+        Sha256Hash btcTxSigHash = btcTx.hashForWitnessSignature(FIRST_INPUT_INDEX, redeemScript, prevValue,
+            BtcTransaction.SigHash.ALL, false);
+
+        // sign with fedKey3
+        signInput(btcTx, fedKey3, FIRST_INPUT_INDEX, btcTxSigHash);
+        TransactionWitness witnessWithFedKey3Signature = btcTx.getWitness(FIRST_INPUT_INDEX);
+
+        // act
+        int sigIndexForFedKey1 = witnessWithFedKey3Signature.getSigInsertionIndex(btcTxSigHash, fedKey1);
+        int sigIndexForFedKey2 = witnessWithFedKey3Signature.getSigInsertionIndex(btcTxSigHash, fedKey2);
+
+        // assert
+        Assert.assertEquals(0, sigIndexForFedKey1);
+        Assert.assertEquals(0, sigIndexForFedKey2);
+    }
+
+    @Test
+    public void getSigInsertionIndex_withWitnessFilledWithSignatures_shouldReturnTheProperIndex() {
+        BtcTransaction btcTx = getBtcTransactionWithBaseWitnessInInput();
+        Sha256Hash btcTxSigHash = btcTx.hashForWitnessSignature(FIRST_INPUT_INDEX, redeemScript, prevValue,
+            BtcTransaction.SigHash.ALL, false);
+        TransactionWitness transactionWitness = btcTx.getWitness(FIRST_INPUT_INDEX);
+
+        int witnessPushCount = transactionWitness.getPushCount();
+        int sigsSuffixCount = p2shOutputScript.getSigsSuffixCount();
+
+        // the pushes that should have the signatures
+        // are between first one (empty byte for checkmultisig bug)
+        // and second to last one (op_notif + redeem script)
+        byte[] secondToLastPush = transactionWitness.getPush(witnessPushCount - sigsSuffixCount - 1);
+        byte[] emptyByte = new byte[]{};
+
+        int i = 0;
+        while(Arrays.equals(secondToLastPush, emptyByte)) {
```

Comment:
This is a bit strange. Why not simply the number of signatures required from the redeem script and iterate that value

Reply:
I tried this using `createBaseWitnessThatSpendsFromErpRedeemScript` that gets 

`int numberOfSignaturesRequiredToSpend = redeemScript.getNumberOfSignaturesRequiredToSpend();`
Which is 5 signatures. However, the check in the TransactionWitness is the following

```
        byte[] secondToLastPush = getPush(totalPushes - sigsSuffixCount - 1);
        boolean hasMissingSigs = Arrays.equals(secondToLastPush, emptyByte);
```

And totalPushes - sigsSuffixCount - 1 is 6

I am losing something?